services/azure-servicebus/src/main/java/com/microsoft/windowsazure/services/servicebus/implementation/ServiceBusRestProxy.java

```diff
@@ -552,34 +284,24 @@ public ListTopicsResult listTopics() throws ServiceException {
     }
 
     @Override
-    public ListSubscriptionsResult listSubscriptions(String topicName)
-            throws ServiceException {
+    public ListSubscriptionsResult listSubscriptions(String topicName) throws ServiceException {
         return listSubscriptions(topicName, ListSubscriptionsOptions.DEFAULT);
     }
 
     @Override
-    public ListRulesResult listRules(String topicName, String subscriptionName)
-            throws ServiceException {
+    public ListRulesResult listRules(String topicName, String subscriptionName) throws ServiceException {
         return listRules(topicName, subscriptionName, ListRulesOptions.DEFAULT);
     }
 
     @Override
-    public void renewQueueLock(String queueName, String messageId,
-            String lockToken) throws ServiceException {
-        ClientResponse clientResponse = getResource().path(queueName)
-                .path("messages").path(messageId).path(lockToken)
-                .post(ClientResponse.class, "");
-        PipelineHelpers.throwIfNotSuccess(clientResponse);
+    public void renewQueueLock(String queueName, String messageId, String lockToken) throws ServiceException {
+        this.async().renewQueueLock(queueName, messageId, lockToken).toBlocking().single();
```

Comment:
I think it's possible to write static T resolve(Observable<T> observable) { return observable.toBlocking().single(); }